typescript/sdk/src/token/adapters/EvmTokenAdapter.ts

```diff
@@ -251,13 +257,69 @@ export class EvmHypSyntheticAdapter
     return this.getTotalSupply();
   }
 
-  async quoteTransferRemoteGas(
-    destination: Domain,
-  ): Promise<InterchainGasQuote> {
-    const gasPayment = await this.contract.quoteGasPayment(destination);
-    // If EVM hyp contracts eventually support alternative IGP tokens,
-    // this would need to determine the correct token address
-    return { amount: BigInt(gasPayment.toString()) };
+  async quoteTransferRemoteGas({
+    destination,
+    recipient,
+    amount,
+  }: QuoteTransferRemoteParams): Promise<InterchainGasQuote> {
+    const contractVersion = await this.contract.PACKAGE_VERSION();
+
+    const hasQuoteTransferRemote =
+      compareVersions(
+        contractVersion,
+        QUOTE_TRANSFER_REMOTE_CONTRACT_VERSION,
+      ) >= 0;
+
+    // Version does not support quoteTransferRemote defaulting to quoteGasPayment
+    if (!hasQuoteTransferRemote) {
```

Comment:
nit: perhaps try to avoid negating the boolean (`!`) so that it's easier to reason about for the reader

```suggestion
    if (useQuoteGasPayment) {
```